test/extensions/filters/common/lua/lua_test.cc

```diff
@@ -157,6 +159,55 @@ TEST_F(LuaTest, MarkDead) {
   lua_gc(cr1->luaState(), LUA_GCCOLLECT, 0);
 }
 
+class ThreadSafeTest : public testing::Test {
+public:
+  ThreadSafeTest()
+      : api_(Api::createApiForTest()), main_dispatcher_(api_->allocateDispatcher("main")),
+        worker_dispatcher_(api_->allocateDispatcher("worker")) {}
+
+  // Use real dispatchers to verify that callback functions can be executed correctly.
+  Api::ApiPtr api_;
+  Event::DispatcherPtr main_dispatcher_;
+  Event::DispatcherPtr worker_dispatcher_;
+  ThreadLocal::InstanceImpl tls_;
+
+  std::unique_ptr<ThreadLocalState> state_;
+};
+
+// Test whether ThreadLocalState can be safely released.
+TEST_F(ThreadSafeTest, StateDestructedBeforeWorkerRun) {
```

Comment:
Does this test fail without the code change in this PR?

Reply:
Yes, it will be a crash.

Reply:
Thanks for verifying!